dbms/src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -2940,37 +2949,29 @@ BlockInputStreams StorageReplicatedMergeTree::read(
     * 2. Do not read parts that have not yet been written to the quorum of the replicas.
     * For this you have to synchronously go to ZooKeeper.
     */
-    Int64 max_block_number_to_read = 0;
+    PartitionIdToMaxBlock max_added_blocks_with_quorum;
     if (settings.select_sequential_consistency)
     {
         auto zookeeper = getZooKeeper();
 
-        String last_part;
-        zookeeper->tryGet(zookeeper_path + "/quorum/last_part", last_part);
-
-        if (!last_part.empty() && !data.getActiveContainingPart(last_part))    /// TODO Disable replica for distributed queries.
-            throw Exception("Replica doesn't have part " + last_part + " which was successfully written to quorum of other replicas."
-                " Send query to another replica or disable 'select_sequential_consistency' setting.", ErrorCodes::REPLICA_IS_NOT_IN_QUORUM);
+        String added_parts_str;
+        zookeeper->tryGet(zookeeper_path + "/quorum/last_part", added_parts_str);
 
-        if (last_part.empty())  /// If no part has been written with quorum.
-        {
-            String quorum_str;
-            if (zookeeper->tryGet(zookeeper_path + "/quorum/status", quorum_str))
-            {
-                ReplicatedMergeTreeQuorumEntry quorum_entry;
```

Comment:
The logic in this if branch is lost (the case when the first quorum insert is in progress). Why?